### PR TITLE
Update changelog for 4.1.0 release

### PR DESCRIPTION
**Pull request by LuyaLiu**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Update changelog for 4.1.0 release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
 
+ ### App Center Crashes
+
+ #### Android
+
+* **[Fix]** Fix removing throwable files after rewriting error logs due to small database size.
+
+#### iOS
+
+* **[Fix]** Fix nullability for attachments methods in crashes delegate. `nil` is valid return value now (4.0.0 regression).
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:36]: please move it to here https://github.com/microsoft/appcenter-sdk-dotnet/pull/1484/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR15

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
 
-## Version 4.0.1 (under development)
+## Version 4.1.0
 
 ### AppCenter
 
 * **[Feature]** Add a `SetMaxStorageSizeAsync` API which allows setting a maximum size limit on the local SQLite storage. The default value is 10MiB.
 
+#### iOS
+
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:10]: please move it to [here](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1484/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR15)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,21 +1,39 @@
...
+* **[Fix]** Fix a crash when the app is trying to open the system settings screen from the background.
+* **[Fix]** Fix browser opening when using a private distribution group on Android 11.
+
 #### iOS/Android
 
  * **[Feature]** Add `NoReleaseAvailable` callback to distribute listener.
 
+ ### App Center Crashes
+
+ #### Android
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:30]: please move it to [here](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1484/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR19)

**Review by AnastasiaKubova — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (MatkovIvan) [inline CHANGELOG.md:36]: It's not related to .NET

**Pull request merged**